pkg/network/ebpf/c/protocols/protocol-classification-helpers.h

```diff
@@ -14,13 +14,52 @@ long bpf_skb_load_bytes(const void *skb, u32 offset, void *to, u32 len) {return
 #endif
 
 #define CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, min_buff_size)   \
-        if (buf_size < min_buff_size) {                                     \
-            return false;                                                   \
-        }                                                                   \
+    if (buf_size < min_buff_size) {                                         \
+        return false;                                                       \
+    }                                                                       \
                                                                             \
-        if (buf == NULL) {                                                  \
-            return false;                                                   \
-        }                                                                   \
+    if (buf == NULL) {                                                      \
+        return false;                                                       \
+    }                                                                       \
+
+static __inline int32_t read_big_endian_int32(const char* buf) {
+    int32_t val;
+    bpf_probe_read_kernel(&val, 4, (void*)buf);
+    return bpf_ntohl(val);
+}
+
+static __inline int16_t read_big_endian_int16(const char* buf) {
+    int16_t val;
+    bpf_probe_read_kernel(&val, 2, (void*)buf);
+    return bpf_ntohs(val);
+}
+
+// Checking if the buffer represents kafka message
+static __always_inline bool is_kafka(const char* buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, KAFKA_MIN_SIZE)
+
+    const int32_t message_size = read_big_endian_int32(buf);
+    size_t offset = sizeof(int32_t);
```

Comment:
Consider setting the offset to the size of the variable (`sizeof(message_size)`)

Reply:
done